### PR TITLE
8305959: x86: Improve itable_stub

### PR DESCRIPTION
**Pull request by dchuyko**
This is a performance backport from 21. Original patch was applied mostly cleanly. Few bits required slight tuning. 

Straightforward replacements were made in in macroAssembler_x86.cpp because of missing JDK-8308396 refactoring (21):
in_bytes(itableMethodEntry::method_offset()) to itableMethodEntry::method_offset_in_bytes(),
in_bytes(itableOffsetEntry::interface_offset()) to itableOffsetEntry::interface_offset_in_bytes(),
in_bytes(itableOffsetEntry::offset_offset()) to itableOffsetEntry::offset_offset_in_bytes().

One java import hunk was manually redone in the benchmark class.

Testing: linux x86 tier1,2; the benchmark shows improvements similar to original results.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] [JDK-8305959](https://bugs.openjdk.org/browse/JDK-8305959) needs maintainer approval

### Issue
 * [JDK-8305959](https://bugs.openjdk.org/browse/JDK-8305959): x86: Improve itable_stub (**Enhancement** - P4 - Approved)

### Reviewers
 * [Volker Simonis](https://openjdk.org/census#simonis) (@simonis - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk17u-dev.git pull/2654/head:pull/2654` \
`$ git checkout pull/2654`

Update a local copy of the PR: \
`$ git checkout pull/2654` \
`$ git pull https://git.openjdk.org/jdk17u-dev.git pull/2654/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 2654`

View PR using the GUI difftool: \
`$ git pr show -t 2654`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk17u-dev/pull/2654.diff">https://git.openjdk.org/jdk17u-dev/pull/2654.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk17u-dev/pull/2654#issuecomment-2205631655)

### CODE REVIEW

File: src/hotspot/cpu/x86/vtableStubs_x86_64.cpp [hunk 1]
```diff
@@ -176,10 +176,12 @@ VtableStub* VtableStubs::create_itable_stub(int itable_index) {
   // (various calling sequences use r[cd]x, r[sd]i, r[89]; stay away from them)
```
Comment (simonis) [inline src/hotspot/cpu/x86/vtableStubs_x86_64.cpp:176]: We forgot to add `r14` to the comment about used registers. But that was already the case in the original change, so I think it's fine too leave it as is.

**Review by simonis — APPROVED** [review]
Thanks for doing this downport. Looks good to me.

**Comment by dchuyko** [comment]
/approval request This is a backport from 21 for x86_64/32. It improves itable_stubs performace which can have notable impact in some applications.

Original patch was applied mostly cleanly, except few straightforward replacements.

InterfaceCalls benchmark shows improvements similar to original results.

A meaningful part of the change is the different code generation in VtableStubs::create_itable_stub(), the current version of which works as of JDK 21, so the risk is low and any potential backout is easy.

Testing: linux x86_64 tier1,2.

File: src/hotspot/cpu/x86/vtableStubs_x86_64.cpp (hunk 1 above)
Comment (dchuyko) [reply src/hotspot/cpu/x86/vtableStubs_x86_64.cpp:176]: Thank you for the review. Yes, it looks right to add comments upstream.

**Comment by phohensee** [comment]
Looks like the macos builds are hung looking for a host on which to run.

**Pull request closed without merging**